modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/nativelib/ArtifactCopyNativeLibraryFinder.java

```diff
@@ -115,24 +113,29 @@ private File copyNativeLibrary(String name, String libraryPath) {
 
   private String findLibraryLocally(String name) {
     String nativeLibName = System.mapLibraryName(name);
-    String extension = FilenameUtils.getExtension(nativeLibName);
-    Optional<URL> nativeLib =
-        stream(urls).filter((URL url) -> {
-          String fullPath;
-          try {
-            fullPath = url.toURI().toString();
-          } catch (URISyntaxException e) {
-            return false;
-          }
-          String baseName = getBaseName(fullPath);
-          return baseName.contains(name) && (fullPath.endsWith(extension) || IS_OS_MAC && fullPath.endsWith(JNILIB_EXTENSION));
-        }).findFirst();
-
-    if (nativeLib.isPresent()) {
-      return nativeLib.get().getFile();
-    } else {
-      return null;
+    String extension = getExtension(nativeLibName);
+    String localLibraryEqualName = null;
+    String localLibraryContainsName = null;
+
+    for (URL url : this.urls) {
+      String fullPath;
+      try {
+        fullPath = url.toURI().toString();
+      } catch (URISyntaxException e) {
+        throw new MuleRuntimeException(e);
+      }
+      boolean endsWithExtension = fullPath.endsWith(extension) || IS_OS_MAC && fullPath.endsWith(JNILIB_EXTENSION);
+      if (localLibraryEqualName == null && getLibraryBaseName(fullPath).equals(name) && endsWithExtension) {
```

Comment:
You are checking the same condition (endsWithExtension) on both branches of the conditional block. You need to simplify this code block.